external/audiocodec/streaming/player.c

```diff
@@ -29,615 +29,555 @@
+		};
+
+		static const int kBitrateV1L3[] = {
+			32, 40, 48, 56, 64, 80, 96, 112,
+			128, 160, 192, 224, 256, 320
+		};
+
+		static const int kBitrateV2[] = {
+			8, 16, 24, 32, 40, 48, 56, 64,
+			80, 96, 112, 128, 144, 160
+		};
+
+		int bitrate;
+		if (version == 3 /* V1 */) {
+			bitrate = (layer == 2 /* L2 */)
+					  ? kBitrateV1L2[bitrate_index - 1]
+					  : kBitrateV1L3[bitrate_index - 1];
+		} else {
+			// V2 (or 2.5)
+			bitrate = kBitrateV2[bitrate_index - 1];
+		}
 
-    if (layer == 3)
-	{
-        // layer I
-        static const int kBitrateV1[] = {
-            32, 64, 96, 128, 160, 192, 224, 256,
-            288, 320, 352, 384, 416, 448
-        };
-
-        static const int kBitrateV2[] = {
-            32, 48, 56, 64, 80, 96, 112, 128,
-            144, 160, 176, 192, 224, 256
-        };
-
-        int bitrate =
-            (version == 3 /* V1 */)
-                ? kBitrateV1[bitrate_index - 1]
-                : kBitrateV2[bitrate_index - 1];
-
-        if (out_bitrate)
-            *out_bitrate = bitrate;
-
-        *frame_size = (12000 * bitrate / sampling_rate + padding) * 4;
-
-        if (out_num_samples)
-            *out_num_samples = 384;
-    }
-	else
-	{
-        // layer II or III
-        static const int kBitrateV1L2[] = {
-            32, 48, 56, 64, 80, 96, 112, 128,
-            160, 192, 224, 256, 320, 384
-        };
-
-        static const int kBitrateV1L3[] = {
-            32, 40, 48, 56, 64, 80, 96, 112,
-            128, 160, 192, 224, 256, 320
-        };
-
-        static const int kBitrateV2[] = {
-            8, 16, 24, 32, 40, 48, 56, 64,
-            80, 96, 112, 128, 144, 160
-        };
-
-        int bitrate;
-        if (version == 3 /* V1 */) {
-            bitrate = (layer == 2 /* L2 */)
-                ? kBitrateV1L2[bitrate_index - 1]
-                : kBitrateV1L3[bitrate_index - 1];
-
-            if (out_num_samples) {
-                *out_num_samples = 1152;
-            }
-        }
-		else
-		{
-            // V2 (or 2.5)
-
-            bitrate = kBitrateV2[bitrate_index - 1];
-            if (out_num_samples)
-                *out_num_samples = (layer == 1 /* L3 */) ? 576 : 1152;
-        }
-
-        if (out_bitrate)
-            *out_bitrate = bitrate;
-
-        if (version == 3 /* V1 */)
-		{
-            *frame_size = 144000 * bitrate / sampling_rate + padding;
-        }
-		else
-        {
-            // V2 or V2.5
-            size_t tmp = (layer == 1 /* L3 */) ? 72000 : 144000;
-            *frame_size = tmp * bitrate / sampling_rate + padding;
-        }
-    }
-
-    if (out_sampling_rate)
-        *out_sampling_rate = sampling_rate;
-
-    if (out_channels)
-	{
-        int channel_mode = (header >> 6) & 3;
-        *out_channels = (channel_mode == 3) ? 1 : 2;
-    }
+		if (version == 3 /* V1 */) {
+			*frame_size = 144000 * bitrate / sampling_rate + padding;
+		} else {
+			// V2 or V2.5
+			size_t tmp = (layer == 1 /* L3 */) ? 72000 : 144000;
```

Comment:
Instead of these description, why don't you define values?
#define MPEG_VERSION_1 3
#define MPEG_VERSION_2 2
...
#define MPEG_LAYER_1 3
#define MPEG_LAYER_2 2

Then you can remove these annotations and clean code.